JW-1307: make create-builder-template work on Codilime's CI

sodar opened the pull request:
- Moved original roles/vars from previous repository

    ```
    Author: Michał Bilski <michal.bilski@codilime.com>
    Author: Michał Kostrzewa <michal.kostrzewa@codilime.com>
    Author: Wojciech Urbański <wojciech.urbanski@codilime.com>
    ```

- Moved `create-builder-template` pipeline script to Jenkinsfile
- Fix some deployment issues

m-kostrzewa commented:
@sodar which changes are file moves and which are actual changes to code?

wurbanski commented on jenkinsfiles/Jenkinsfile.create-builder-template:
```diff
@@ -0,0 +1,49 @@
+withCredentials([usernamePassword(credentialsId: 'e9f31c41-6d0e-4950-b6d7-1aa7080468aa', passwordVariable: 'VC_PASSWORD', usernameVariable: 'VC_USERNAME')]) {
```
Consider reworking this to follow *declarative pipeline* style from `Jenkinsfile.deploy-builder` in the old repository.
Please. Strongly consider.

wurbanski reviewed (changes requested):
Please show us what changed :).

Commit:
JW-1307: moved original files related to deploying builders

Author: Michał Bilski <michal.bilski@codilime.com>
Author: Michał Kostrzewa <michal.kostrzewa@codilime.com>
Author: Wojciech Urbański <wojciech.urbanski@codilime.com>

Commit:
JW-1307: prepare create-builder-template for running in Codilime CI

- Use `inventory.sample` as a template for `inventory`.
- Change `readTrusted` to `readFile`, since `readTrusted` does not do parameter
  expansion. It caused problems when branch name is passed as a parameter.

sodar commented:
@wurbanski @m-kostrzewa Sorry about the mess with the commits. I have rebased the changes, so that all moved/copied code is in one commit, i.e. the earliest commit in this PR.

I need to test the rest of the changes and refactor Jenkinsfile for create-builder-template so I mark this PR as WIP.

sodar commented:
PR is ready for review. Commit with hash `999986...` contains my changes.

Commit:
JW-1307: move shared drive and jenkins master IP to vars file

Commit:
JW-1307: config generated by createBuilderTemplate script

wurbanski reviewed (approved):
LGTM
